lib/fluent/plugin/out_datadog.rb

```diff
@@ -131,40 +99,35 @@ def write(chunk)
         messages.push "#{api_key} " + Yajl.dump(record) + "\n"
       else
         next unless record.has_key? "message"
-        messages.push "#{api_key} " + record["message"].rstrip() + "\n"
+        messages.push "#{api_key} " + record["message"].strip + "\n"
       end
     end
     send_to_datadog(messages)
-
   end
 
-  def send_to_datadog(data)
+  def send_to_datadog(events)
     @my_mutex.synchronize do
-      retries = 0
-      begin
-        log.trace "Send nb_event=#{data.size} events to Datadog"
-
-        # Check the connectivity and write messages
-        log.info "New attempt to Datadog attempt=#{retries}" if retries > 0
-
-        data.each do |event|
-          log.trace "Datadog plugin: about to send event=#{event}"
-          client.write(event)
-        end
-
-        # Handle some failures
-      rescue Errno::EHOSTUNREACH, Errno::ECONNREFUSED, Errno::ETIMEDOUT, Errno::EPIPE => e
-
-        if retries < @max_retries || max_retries == -1
-          @_socket = nil
-          a_couple_of_seconds = retries ** 2
-          a_couple_of_seconds = 30 unless a_couple_of_seconds < 30
-          retries += 1
-          log.warn "Could not push logs to Datadog, attempt=#{retries} max_attempts=#{max_retries} wait=#{a_couple_of_seconds}s error=#{e.message}"
-          sleep a_couple_of_seconds
-          retry
+      events.each do |event|
+        log.trace "Datadog plugin: about to send event=#{event}"
+        retries = 0
+        begin
+          log.info "New attempt to Datadog attempt=#{retries}" if retries > 0
```

Comment:
If we skip logging the first failure like I suggested below, this should probably be only logged if `retries > 1`